src/traces/funnel/defaults.js

```diff
@@ -0,0 +1,79 @@
+/**
+* Copyright 2012-2019, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var Lib = require('../../lib');
+
+var handleGroupingDefaults = require('../bar/defaults').handleGroupingDefaults;
+var handleText = require('../bar/defaults').handleText;
+var handleXYDefaults = require('../scatter/xy_defaults');
+var attributes = require('./attributes');
+var Color = require('../../components/color');
+
+function supplyDefaults(traceIn, traceOut, defaultColor, layout) {
+    function coerce(attr, dflt) {
+        return Lib.coerce(traceIn, traceOut, attributes, attr, dflt);
+    }
+
+    var len = handleXYDefaults(traceIn, traceOut, layout, coerce);
+    if(!len) {
+        traceOut.visible = false;
+        return;
+    }
+
+    coerce('orientation', (traceOut.x && !traceOut.y) ? 'h' : 'v');
+    coerce('offset');
+    coerce('width');
+
+    coerce('text');
+    coerce('textinfo');
+
+    coerce('hovertext');
+    coerce('hovertemplate');
+
+    handleText(traceIn, traceOut, layout, coerce, false);
+
+    coerce('marker.color', defaultColor);
+    coerce('marker.line.color', Color.defaultLine);
+    coerce('marker.line.width');
+
+    var connectorVisible = coerce('connector.visible');
+    if(connectorVisible) {
+        coerce('connector.fillcolor', Color.addOpacity(defaultColor, 0.5));
```

Comment:
I'm a fan of this, but we should probably double-check with @plotly/plotly_js and @nicolaskruchten

Reply:
This is what makes the fill color the same color as the bar but lighter? This is as requested by me, so I approve :)

Reply:
Shouldn’t we be adding opacity to `marker.color`? So if you set your own color the fill matches it by default?

Reply:
Good call. Fixed in https://github.com/plotly/plotly.js/pull/3817/commits/6c8a661fe4b458915ce8428a765c9a6dd0efd3b1.

Reply:
Oh, good thinking about handling the arrayOk case, I not entirely sure your

```js
        var defaultFillcolor = Color.addOpacity(
            Lib.isArrayOrTypedArray(markerColor) ? markerColor[0] : markerColor, 0.5
        );
        coerce('connector.fillcolor', defaultFillcolor);
```

is ideal. How hard would it be to add `arrayOk:true` support for `connector.fillcolor`?

Reply:
Alternatively, we could not set the `connector.fillcolor` default in the case where `marker.color` is an array, and then turn 

https://github.com/plotly/plotly.js/blob/6a3a13c6f20348a5637eaa87a81d07eb28125145/src/traces/funnel/style.js#L41-L43

into

```js
        gTrace.selectAll('.regions').each(function(di) {
            d3.select(this).selectAll('path')
               .style('stroke-width', 0)
               .call(Color.fill, trace.connector.fillcolor || di.mc);
        });
```

Reply:
that's weird though - the connector links bars `i` and `i+1`, so which `marker.color` do we take, `i` or `i+1`? I mean, I'd be OK with accepting an array of length `n-1` (though... `_length` & transforms 🤦‍♂), but pulling `n-1` items out of a length-`n` array seems like a bad idea.

I'd fall back on something like `rgba(0,0,0,0.5)` as the default when `marker.color` is an array. Not even `defaultColor`, as the user has already said they don't want that.

Reply:
Totally OK to change this to the way it should be. 
However; one may argue that the current behaviour is consistent with the way labels are presented. Also since funnel charts often have greater values in their very first elements (as the whole) it may be OK to present them this way.    
https://github.com/plotly/plotly.js/pull/3817/commits/6c8a661fe4b458915ce8428a765c9a6dd0efd3b1?short_path=4dff010#diff-4dff0100e1e69e85f5abd051fbd5c9f2 
![Screenshot from 2019-05-01 20-06-03](https://user-images.githubusercontent.com/33888540/57050975-20cb8080-6c4d-11e9-97ce-2a1acbe322ea.png)

Reply:
> I'd fall back on something like `rgba(0,0,0,0.5)` as the default when `marker.color` is an array.

I'm a fan of this :ok_hand:

Reply:
This is fixed here: 
https://github.com/plotly/plotly.js/blob/c29b58d25b1d61749ddeded441420979f8ad8d13/src/traces/funnel/defaults.js#L64-L68
Note it was also required to take into account the `trace.opacity`.
So here is the new baseline https://github.com/plotly/plotly.js/blob/c29b58d25b1d61749ddeded441420979f8ad8d13/test/image/baselines/funnel_custom.png
which renders https://github.com/plotly/plotly.js/blob/e968da68abbc70c67a0b08f646d266a2e1754dd2/test/image/mocks/funnel_custom.json

Reply:
Nicely done :trophy: